cpp/src/join/hash_join.cu

```diff
@@ -81,7 +81,7 @@ void build_join_hash_table(cudf::table_view const& build,
   CUDF_EXPECTS(0 != build_table_ptr->num_columns(), "Selected build dataset is empty");
   CUDF_EXPECTS(0 != build_table_ptr->num_rows(), "Build side table has no rows");
 
-  row_hash hash_build{*build_table_ptr};
+  row_hash hash_build{nullate::YES{}, *build_table_ptr};
```

Comment:
I'd actually like to see this made to dynamic instead. I think this always assumed `has_nulls == true` to avoid compiling two kernels, but now we can just make it dynamic.

Reply:
Sounds good. 
This looks like a very involved change over several files since the `row_hash` type used here is actually defined in header file [join_common_utils.hpp](https://github.com/rapidsai/cudf/pull/9623/files#diff-fe28a4362c6d7a5d462be7c23b9d0a2aab1d8e61cc1a4515f863652c53f01f9fR50)
This type appears to be used in `hash_join.cu`, `hash_join.cuh`, and `semi_join.cu`  right now.
So I would prefer to make this change in a follow on PR.

Reply:
Yeah, follow up PR implied :)